website_portal_medical_patient/tests/test_medical_patient.py

```diff
@@ -10,12 +10,24 @@ class TestMedicalPatient(TransactionCase):
     def setUp(self):
         super(TestMedicalPatient, self).setUp()
         self.patient_1 = self.env.ref(
-            'website_portal_medical_patient.medical_patient_1'
+            'medical.medical_patient_patient_1'
         )
+        self.portal_user = self.env.ref(
+            'portal.demo_user0'
+        )
+        self.Patient = self.env['medical.patient']
```

Comment:
This is not a blocker, but there's no need to create these two attributes in ``setUp`` since they're only being used in one test.

Reply:
Regardless of how many tests there are I usually put these things in the setUp method for future use if/when needed.

Reply:
The downside to this approach is that the logic will run once for every test case, rather than just the case that requires it, making the test suite slower. Also, it doesn't help with code readability because the test case logic is split across the file.

In this scenario, the difference is trivial, but in situations with a lot of test cases or more complex logic than just retrieving some recordsets, it may be significant, so I try to move code into ``setUp`` only once it's required in multiple test cases.

Reply:
Fair point. As you said yeah if there's a lot of tests it can be an issue. Like if there's logic for 1 test in the setUp and all the other tests aren't using that logic, that wouldn't make sense haha.